src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -49,16 +71,17 @@ subroutine init_atm_static(mesh, dims, configs)
 
  integer:: isice_lu,iswater_lu,ismax_lu
 
- integer:: nx,ny,nz
- integer:: endian,isigned,istatus,wordsize
+ integer(c_int):: nx,ny,nz
+ integer(c_int):: endian,isigned,istatus,wordsize
  integer:: i,j,k
  integer:: iCell,iEdge,iVtx,iPoint,iTileStart,iTileEnd,jTileStart,jTileEnd
  integer,dimension(5) :: interp_list
  integer,dimension(:),allocatable  :: nhs
  integer,dimension(:,:),allocatable:: ncat
       
- real(kind=4):: scalefactor
- real(kind=4),dimension(:,:,:),allocatable:: rarray
+ real(kind=c_float):: scalefactor
+ real(c_float),dimension(:,:,:),pointer,contiguous :: rarray
```

Comment:
I'm not sure if the Fortran standard expresses a preference for either style (e.g., `real(kind=c_float)` versus `real(c_float)`), but we should declare `scalefactor` and `rarray` consistently.